parser/src/parser/tests/trace_columns.rs

```diff
@@ -25,20 +28,50 @@ fn trace_columns_main_and_aux() {
     trace_columns:
         main: [clk, fmp, ctx]
         aux: [rc_bus, ch_bus]";
-    let expected = Source(vec![TraceCols(TraceCols {
+    let expected = Source(vec![Trace(Trace {
         main_cols: vec![
-            Identifier("clk".to_string()),
-            Identifier("fmp".to_string()),
-            Identifier("ctx".to_string()),
+            TraceCols::new(Identifier("clk".to_string()), 1),
+            TraceCols::new(Identifier("fmp".to_string()), 1),
+            TraceCols::new(Identifier("ctx".to_string()), 1),
         ],
         aux_cols: vec![
-            Identifier("rc_bus".to_string()),
-            Identifier("ch_bus".to_string()),
+            TraceCols::new(Identifier("rc_bus".to_string()), 1),
+            TraceCols::new(Identifier("ch_bus".to_string()), 1),
         ],
     })]);
     build_parse_test!(source).expect_ast(expected);
 }
 
+#[test]
+fn trace_columns_groups() {
+    let source = "
+    trace_columns:
+        main: [clk, fmp, ctx, a[3]]
+        aux: [rc_bus, b[4], ch_bus]
+    transition_constraints:
+        enf a[1]' = 1";
```

Comment:
We should probably also include an example of a trace column that isn't indexed into (e.g. `clk`) to ensure that it ends up with the correct index (e.g. as `TraceAccess::new(Identifier("clk".to_string()), 0)`)